lib/surgex/parser/parsers/id_parser.ex

```diff
@@ -3,19 +3,52 @@ defmodule Surgex.Parser.IdParser do
 
   alias Surgex.Parser.IntegerParser
 
-  @type errors :: :invalid_identifier | IntegerParser.errors()
+  @type errors :: :invalid_identifier | :invalid_max | IntegerParser.errors()
+  @type numeric_id_types :: :int | :integer | :bigint | :biginteger | :serial | :bigserial
+  @type option :: {:max, integer() | numeric_id_types()}
 
-  @spec call(term()) :: {:ok, integer | nil} | {:error, errors}
-  def call(nil), do: {:ok, nil}
-  def call(""), do: {:ok, nil}
+  # This parser already validates that ids cannot be below zero
+  # therefore we don't handle the minimum case
+  # See reference here for numeric types: https://www.postgresql.org/docs/current/datatype-numeric.html
 
-  def call(input) when is_binary(input) do
-    case IntegerParser.call(input) do
-      {:ok, integer} when integer > 0 -> {:ok, integer}
-      {:ok, _invalid_integer} -> {:error, :invalid_identifier}
-      {:error, reason} -> {:error, reason}
+  @int8_max 9_223_372_036_854_775_807
+  @int4_max 2_147_483_647
+
+  @numeric_type_to_max_size %{
+    bigint: @int8_max,
+    biginteger: @int8_max,
+    bigserial: @int8_max,
+    int: @int4_max,
+    integer: @int4_max,
+    serial: @int4_max
+  }
+
+  @spec call(term(), [option()]) :: {:ok, integer | nil} | {:error, errors}
+  def call(input, opts \\ [])
+  def call(nil, _opts), do: {:ok, nil}
+  def call("", _opts), do: {:ok, nil}
+
+  def call(input, opts) when is_binary(input) do
+    max = Keyword.get(opts, :max)
+
+    with {:ok, max} <- parse_max(max) do
+      case IntegerParser.call(input, max: max) do
+        {:ok, integer} when integer > 0 -> {:ok, integer}
+        {:ok, _invalid_integer} -> {:error, :invalid_identifier}
+        {:error, reason} -> {:error, reason}
+      end
```

Comment:
```suggestion
    with {:ok, max} <- parse_max(max) do
      case IntegerParser.call(input, max: max, min: 1) do
        {:ok, integer} -> {:ok, integer}
        {:error, reason} -> {:error, reason}
      end
```
Integer parser also has a `min` option, so you don't need to check it manually here.

Reply:
I like this suggestion.

I have a slight concern though that it might break existing code that relies on that status, and arguably `invalid_identifier` is more descriptive than `out_of_range`. So I'll modify that slightly.